cmd/metacache-server-pool.go

```diff
@@ -419,6 +419,26 @@ func (z *erasureServerPools) listAndSave(ctx context.Context, o *listPathOptions
 		o.debugln("listAndSave: listing", o.ID, "finished with ", err)
 	}(*o)
 
+	// Keep alive while initial request is running.
+	go func(ctx context.Context, rpc *peerRESTClient, meta metacache) {
+		// Continuously update while we wait.
+		t := time.NewTicker(metacacheMaxClientWait / 10)
+		defer t.Stop()
+		select {
```

Comment:
@klauspost Is this missing a `for {` wrapping the select, to keep it running continuously?

Reply:
yes

Reply:
Well, since I copy+pasted....

Reply:
Good spot! Moving the shared code to a function.